python/dask_cudf/dask_cudf/io/parquet.py

```diff
@@ -292,24 +292,45 @@ def write_partition(
             preserve_index = True
         if partition_on:
             md = write_to_dataset(
-                df,
-                path,
+                df=df,
+                root_path=path,
+                compression=compression,
                 filename=filename,
                 partition_cols=partition_on,
                 fs=fs,
                 preserve_index=preserve_index,
                 return_metadata=return_metadata,
-                **kwargs,
+                statistics=kwargs.get("statistics", "ROWGROUP"),
```

Comment:
If these defaults match those of the `write_to_dataset` function, it's perfectly fine to pass `**kwargs` through here rather than re-define the defaults. We just want to guard against unrecognized/unsupported names in `**kwargs` making their way all the way down the call stack and being silently ignored (without warnings/errors) if the user expects them to do something.

Alternatively you can use `kwargs.pop("key", "default")` to extract these values before calling `write_to_dataset` and assert that all the `**kwargs` were used -- it should be empty after popping all the known keys. 😉

Reply:
So the thing here is, the method signature is not guaranteed to match to that of `write_to_dataset`, because the upstream API signature is:
```
def write_partition(cls, df, path, fs, filename, **kwargs):
```

To perform validation will be a blocker here because dask sometimes sends a few parameters like `custom_metadata`, etc which aren't relevant to us. Hence doing this process of filtration at this level.

Reply:
Sounds like you have more context here -- I'll defer to your judgment.